gdpr_assist/models.py

```diff
@@ -246,33 +278,278 @@ def _cast_class(cls, model, privacy_meta):
+                |-> pollsession_set = (PollSession [set_field]):
+                    |-> outcome_assessments = (OutcomeAssessment [set_field]):
+                        |-> score
+                        |-> comment
+                    |-> pollresponse_set = (PollResponse [set_field]):
+                        |-> response
+
+        """
+        ABANDON_AFTER_N_LEVELS = 10
+        BASE_PREFIX = "    "
+        if len(prefix) > len(BASE_PREFIX) * ABANDON_AFTER_N_LEVELS:
+            return "ERROR: shouldn't go %s levels deep, check for loops." % ABANDON_AFTER_N_LEVELS
+        res = ''
+        if doprint:
+            res += cls.__name__ + ":\n"
+
+        privacy_meta_model = cls._privacy_meta
+
+        flat_fields = privacy_meta_model.fields
+        if flat_fields:
+            for f in flat_fields:
+                res += "%s|-> %s\n" % (prefix, f)
+
+        for fk in privacy_meta_model.fk_fields:
+
+            try:
+                f = getattr(cls, fk)
+                if hasattr(f, 'field'):
+                    f = f.field
+                elif hasattr(f, 'rel'):
+                    f = f.rel
+                child_model = f.related_model
+                res += "%s|-> %s = (%s [fk]):\n" % (prefix, fk, child_model.__name__)
+
+                res += child_model.get_anonymization_tree(prefix=BASE_PREFIX + prefix, doprint=False)
+            except AttributeError as e:
+                print(e)
+                res += 'ERROR: ' + str(e)
+
+        for set_field in privacy_meta_model.set_fields:
+            res += "%s|-> %s" % (prefix, set_field)
+
+            try:
+                f = getattr(cls, set_field)
+                if hasattr(f, "objects"):
+                    child_model = f.objects.model
+                elif hasattr(f, "rel"):
+                    child_model = f.rel.related_model
+                res += " = (%s [set_field]):\n" % child_model.__name__
+                res += child_model.get_anonymization_tree(prefix=BASE_PREFIX + prefix, doprint=False)
+            except AttributeError as e:
+                print(e)
+                res += 'ERROR: ' + str(e)
+
+        if doprint:
+            print(res)
+        return res
+
     class Meta:
         abstract = True
 
 
+@python_2_unicode_compatible
+class RetentionPolicyItem(PrivacyModel):
+    description = models.CharField(default="", max_length=255)
+    start_date = models.DateTimeField(null=True, blank=False)
+    updated_at = models.DateTimeField(auto_now=True)
+    policy_length = models.DurationField(null=True, blank=True)  # corresponds to a datetime.timedelta
+    retention_policy = None  # it would inherit this from PrivacyModel, but it doesn't make sense here
+
+    @property
+    def target_entities(self):
+        objs = [
+            getattr(self, x).all()
+            for x in self._meta.fields_map.keys() if x.startswith("target")
+        ]
+        res = []
+        for o in objs:
+            res += list(o)
+
+        return res
```

Comment:
Am I missing something here?

```suggestion
        return objs
```